modules/proc_quay-bridge-operator.adoc

```diff
@@ -0,0 +1,228 @@
+  type: ClusterIP
+----
+
+. Create the webhook service as follows:
++
+```
+$ oc create -f quay-webhook.yaml
+```
+. Download the
+link:https://github.com/redhat-cop/quay-openshift-registry-operator/blob/master/hack/webhook-create-signed-cert.sh[webhook-create-signed-cert.sh]
+script, so you can use it to generate a certificate signed by a Kubernetes certificate authority.
+
+. Execute the following command to request the certificate:
++
+```
+$ ./webhook-create-signed-cert.sh --namespace openshift-operators \
+   --secret quay-bridge-operator-webhook-certs \
+   --service quay-bridge-operator
+```
+
+. Execute the following command to retrieve the CA and format the result as a single line
+so that it can be entered into the MutatingWebhookConfiguration resource:
++
+```
+$ oc get configmap -n kube-system \
+   extension-apiserver-authentication \
+   -o=jsonpath='{.data.client-ca-file}' | base64 | tr -d '\n'
+```
+
+. Replace the ${CA_BUNDLE} variable in the following MutatingWebhookConfiguration YAML:
++
+.quay-mutating-webhook.yaml
+[source,yaml]
+----
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: quay-bridge-operator
+webhooks:
+  - name: quayintegration.redhatcop.redhat.io
+    clientConfig:
+      service:
+        namespace: openshift-operators
+        name: quay-bridge-operator
+        path: "/admissionwebhook"
+      caBundle: "${CA_BUNDLE}" <1>
+    rules:
+    - operations:  [ "CREATE" ]
+      apiGroups: [ "build.openshift.io" ]
+      apiVersions: ["v1" ]
+      resources: [ "builds" ]
+    failurePolicy: Fail
+----
+<1> Replace ${CA_BUNDLE} with the output of the previous step. It will appear as one long line that you copy and paste to replace ${CA_BUNDLE}.
+
+. Create the MutatingWebhookConfiguration as follows:
++
+```
+$ oc create -f quay-mutating-webhook.yaml
+```
++
+Until the operator is running, new requests for builds will fail
+since the webserver the MutatingWebhookConfiguration invokes is
+not available and a proper is response is required in order for
+the object to be persisted in etcd.
+
+. Go to the OpenShift console and install the Quay Bridge Operator as follows:
++
+* Select OperatorHub and search for Quay Bridge Operator.
++
+* Select Install
++ 
+* Choose Installation Mode (all namespaces), Update Channel,
+and Approval Strategy (Automatic or Manual).
++
+* Select Subscribe
++
+
+. Create the custom resource (CR) called `QuayIntegration`. For example:
++
+.quay-integration.yaml
+[source,yaml]
+----
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: QuayIntegration
+metadata:
+  name: example-quayintegration
+spec:
+  clusterID: openshift  <1>
+  credentialsSecretName: openshift-operators/quay-integration <2>
+  quayHostname: https://<QUAY_URL>   <3>
+  whitelistNamespaces: <4>
+    - default
+  insecureRegistry: false <5>
+----
+<1> The clusterID value should be unique across the entire ecosystem.
+This value is optional and defaults to openshift.
```

Comment:
The clusterID should not collision if multiple OCP clusters use the same Quay as registry